elastic/kibana
# [Response Ops][Alerting] Adding ability to run actions for backfill rule runs

Resolves https://github.com/elastic/response-ops-team/issues/251


## Note

This PR includes some saved object schema changes that I will pull out into their own separate PR in order to perform an intermediate release. I wanted to make sure all the schema changes made sense in the overall context of the PR before opening those separate PRs.

Update: PR for intermediate release here: https://github.com/elastic/kibana/pull/203184 (Merged)

## Summary

Adds ability to run actions for backfill rule runs.

- Updates schedule backfill API to accept `run_actions` parameter to specify whether to run actions for backfill.
- Schedule API accepts any action where `frequency.notifyWhen === 'onActiveAlert'`. If a rule has multiple actions where some are `onActiveAlert` and some are `onThrottleInterval`, the invalid actions will be stripped and a warning returned in the schedule response but valid actions will be scheduled.
- Connector IDs are extracted and stored as references in the ad hoc run params saved object
- Any actions that result from a backfill task run are scheduled as low priority tasks

## To Verify

1. Create a detection rule. Make sure you have some past data that the rule can run over in order to generate actions. Make sure you add actions to the rule. For testing, I added some conditional actions so I could see actions running only on backfill runs using `kibana.alert.rule.execution.type: "manual"`. Create actions with and without summaries.
2. Schedule a backfill either directly via the API or using the detection UI. Verify that actions are run for the backfill runs that generate alerts.
